lib/file-list.js

```diff
@@ -16,127 +13,56 @@ const helper = require('./helper')
     this._preprocess = Promise.promisify(preprocess)
-    this._autoWatchBatchDelay = autoWatchBatchDelay
 
-    // The actual list of files
     this.buckets = new Map()
 
-    // Internal tracker if we are refreshing.
-    // When a refresh is triggered this gets set
-    // to the promise that `this._refresh` returns.
-    // So we know we are refreshing when this promise
-    // is still pending, and we are done when it's either
-    // resolved or rejected.
     this._refreshing = Promise.resolve()
 
-    // Emit the `file_list_modified` event.
-    // This function is debounced to the value of `autoWatchBatchDelay`
-    // to avoid reloading while files are still being modified.
     const emit = () => {
       this._emitter.emit('file_list_modified', this.files)
     }
 
-    const debouncedEmit = _.debounce(emit, this._autoWatchBatchDelay)
+    const debouncedEmit = _.debounce(emit, autoWatchBatchDelay)
     this._emitModified = (immediate) => {
       immediate ? emit() : debouncedEmit()
     }
   }
 
-  // Private Interface
-  // -----------------
-
-  // Is the given path matched by any exclusion filter
-  //
-  // path - String
-  //
-  // Returns `undefined` if no match, otherwise the matching
-  // pattern.
-  _isExcluded (path) {
-    return _.find(this._excludes, (pattern) => mm(path, pattern))
+  _findExcluded (path) {
+    return this._excludes.find((pattern) => mm(path, pattern))
   }
 
-  // Find the matching include pattern for the given path.
-  //
-  // path - String
-  //
-  // Returns the match or `undefined` if none found.
-  _isIncluded (path) {
-    return _.find(this._patterns, (pattern) => mm(path, pattern.pattern))
+  _findIncluded (path) {
+    return this._patterns.find((pattern) => mm(path, pattern.pattern))
   }
 
-  // Find the given path in the bucket corresponding
-  // to the given pattern.
-  //
-  // path    - String
-  // pattern - Object
-  //
-  // Returns a File or undefined
   _findFile (path, pattern) {
     if (!path || !pattern) return
-    if (!this.buckets.has(pattern.pattern)) return
-
-    return _.find(Array.from(this.buckets.get(pattern.pattern)), (file) => {
-      return file.originalPath === path
-    })
+    return this._getFilesByPattern(pattern.pattern).find((file) => file.originalPath === path)
   }
 
-  // Is the given path already in the files list.
-  //
-  // path - String
-  //
-  // Returns a boolean.
   _exists (path) {
-    const patterns = this._patterns.filter((pattern) => mm(path, pattern.pattern))
-
-    return !!_.find(patterns, (pattern) => this._findFile(path, pattern))
+    return !!this._patterns.find((pattern) => mm(path, pattern.pattern) && this._findFile(path, pattern))
   }
 
-  // Check if we are currently refreshing
-  _isRefreshing () {
-    return this._refreshing.isPending()
+  _getFilesByPattern (pattern) {
+    return this.buckets.get(pattern) || []
   }
 
-  // Do the actual work of refreshing
   _refresh () {
-    const buckets = this.buckets
```

Comment:
Are you confident that changing the delayed-update behavior is completely safe? 

In the old code `this.buckets` is not updated along several code paths. Seems like there could be some ways that this results in a change beyond a refactoring.  If so, I would prefer that the refactoring be separate so any behavior change is isolate to make debugging easier.

Reply:
In old code `this.buckets` was assigned to `buckets` variable at the beginning of `_refresh` method. In the whole `_refresh` method `this.buckets` and `buckets` point to the same object, so `buckets` variable seemed redundant to me - it's why I removed it.

`const buckets = this.buckets` is leftover from time when `arrow functions` was not used inside `_refresh` method.